cmd/performance-profile-creator/cmd/root.go

```diff
@@ -128,86 +131,105 @@ func NewRootCommand() *cobra.Command {
 
 	var requiredFlags = []string{
 		"reserved-cpu-count",
-		"mcp-name",
 		"rt-kernel",
 		"must-gather-dir-path",
 	}
 
 	root := &cobra.Command{
 		Use:   "performance-profile-creator",
 		Short: "A tool that automates creation of Performance Profiles",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			if cmd.Flag("info").Changed {
-				infoMode := cmd.Flag("info").Value.String()
-				if err := validateFlag("info", infoMode, validInfoModes); err != nil {
-					return err
-				}
-
-				missingRequiredFlags := checkRequiredFlags(cmd, "must-gather-dir-path")
-				if len(missingRequiredFlags) > 0 {
-					return fmt.Errorf("missing required flags: %s", strings.Join(argNameToFlag(missingRequiredFlags), ", "))
-				}
-
-				mustGatherDirPath := cmd.Flag("must-gather-dir-path").Value.String()
-				cluster, err := getClusterData(mustGatherDirPath)
-				if err != nil {
-					return fmt.Errorf("failed to parse the cluster data: %v", err)
-				}
-
-				clusterInfo := makeClusterInfoFromClusterData(cluster)
-				if infoMode == infoModeJSON {
-					if err := showClusterInfoJSON(clusterInfo); err != nil {
-						return fmt.Errorf("unable to show cluster info %w", err)
-					}
-				} else {
-					showClusterInfoLog(clusterInfo)
-				}
-				return nil
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			err := validateMustGatherDirPath(pcArgs.MustGatherDirPath)
+			if err != nil {
+				return err
 			}
-
+			pcArgs.createForHypershift, err = hypershift.IsHypershift(pcArgs.MustGatherDirPath)
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
 			missingRequiredFlags := checkRequiredFlags(cmd, requiredFlags...)
 			if len(missingRequiredFlags) > 0 {
 				return fmt.Errorf("missing required flags: %s", strings.Join(argNameToFlag(missingRequiredFlags), ", "))
 			}
-
-			mustGatherDirPath := cmd.Flag("must-gather-dir-path").Value.String()
-			cluster, err := getClusterData(mustGatherDirPath)
+			if err := validateProfileCreatorFlags(pcArgs); err != nil {
+				return err
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			nodes, err := listNodesForPool(pcArgs)
 			if err != nil {
-				return fmt.Errorf("failed to parse the cluster data: %v", err)
+				return err
 			}
-
-			profileCreatorArgsFromFlags, err := getDataFromFlags(cmd)
+			nodesHandlers, err := makeNodesHandlers(pcArgs.MustGatherDirPath, pcArgs.NodePoolName, nodes)
```

Comment:
where are `nodesHandlers` being used here ?  is this only for validation (catching only the error) ?

Reply:
Please check the full changes. `nodeHandlers` are used in lines 171 and 177
https://github.com/openshift/cluster-node-tuning-operator/pull/1124/files/9a4aba81c529c6ce074771593be98f40afd2ea6c#diff-baadfd30860582642e86fbeebbe6cb8ca8b7f39266420331fc322cf237e121fcR171 

https://github.com/openshift/cluster-node-tuning-operator/pull/1124/files/9a4aba81c529c6ce074771593be98f40afd2ea6c#diff-baadfd30860582642e86fbeebbe6cb8ca8b7f39266420331fc322cf237e121fcR177